mobile_config/src/mobile_radio_tracker.rs

```diff
@@ -0,0 +1,337 @@
+                _ = interval.tick() => {
+                    if let Err(err) = track_changes(&self.pool, &self.metadata).await {
+                        tracing::error!(?err, "error in tracking changes to mobile radios");
+                    }
+                }
+            }
+        }
+
+        tracing::info!("stopping");
+
+        Ok(())
+    }
+}
+
+async fn track_changes(pool: &Pool<Postgres>, metadata: &Pool<Postgres>) -> anyhow::Result<()> {
+    tracing::info!("looking for changes to radios");
+    let tracked_radios = get_tracked_radios(pool).await?;
+    let all_mobile_radios = get_all_mobile_radios(metadata);
+
+    let updates = identify_changes(all_mobile_radios, tracked_radios).await;
+    tracing::info!("updating in db: {}", updates.len());
+
+    update_tracked_radios(pool, updates).await?;
+    tracing::info!("done");
+
+    Ok(())
+}
+
+async fn identify_changes(
+    all_mobile_radios: impl Stream<Item = MobileRadio>,
+    tracked_radios: HashMap<EntityKey, TrackedMobileRadio>,
+) -> Vec<TrackedMobileRadio> {
+    all_mobile_radios
+        .scan(tracked_radios, |tracked, radio| {
+            let tracked_radio_opt = tracked.remove(&radio.entity_key);
+            async { Some((radio, tracked_radio_opt)) }
+        })
+        .map(|(radio, tracked_radio_opt)| match tracked_radio_opt {
+            Some(tracked_radio) => tracked_radio.update_from_radio(&radio),
+            None => TrackedMobileRadio::new(&radio),
+        })
+        .collect()
+        .await
+}
+
+async fn get_tracked_radios(
+    pool: &Pool<Postgres>,
+) -> anyhow::Result<HashMap<EntityKey, TrackedMobileRadio>> {
+    sqlx::query_as::<_, TrackedMobileRadio>(
+        r#"
+        SELECT 
+            entity_key,
+            hash,
+            last_changed_at,
+            last_checked_at
+        FROM mobile_radio_tracker
+        "#,
+    )
+    .fetch(pool)
+    .try_fold(HashMap::new(), |mut map, tracked_radio| async move {
+        map.insert(tracked_radio.entity_key.clone(), tracked_radio);
+        Ok(map)
+    })
+    .map_err(anyhow::Error::from)
+    .await
+}
+
+fn get_all_mobile_radios(metadata: &Pool<Postgres>) -> impl Stream<Item = MobileRadio> + '_ {
+    sqlx::query_as::<_, MobileRadio>(
+        r#"
+        SELECT
+        	kta.entity_key,
+        	mhi.refreshed_at,
+        	mhi.location::bigint,
+        	mhi.is_full_hotspot::int,
+        	mhi.num_location_asserts,
+        	mhi.is_active::int,
+        	mhi.dc_onboarding_fee_paid::bigint,
+        	mhi.device_type::text,
+        	mhi.deployment_info::text
+        FROM key_to_assets kta
+        INNER JOIN mobile_hotspot_infos mhi ON
+        	kta.asset = mhi.asset
+        WHERE kta.entity_key IS NOT NULL
+        	AND mhi.refreshed_at IS NOT NULL
+    "#,
+    )
+    .fetch(metadata)
+    .filter_map(|result| async move {
+        if let Err(err) = &result {
+            tracing::error!(?err, "error when reading radio metadata");
+        }
+        result.ok()
+    })
```

Comment:
I'd prefer not to hide such errors but rather return an error result and stop updating to avoid the possibility of finishing with an inconsistent result.

Reply:
This is indeed a code smell.. why is this masking the error?

Reply:
we do this in several places where we perform `get_all` operations by services/agents; typically the logic has been to allow the overall system to continue operating in a degraded state, reporting errors as they happen, rather than causing a single bad record in 10s of thousands to cause the entire operation like a cache refresh in the mobile-verifier or rewardable entity distribution to grind to a halt, usually in off hours

Reply:
fair.. the downside is that without looking at the logs (which no client does) you'd end up operating on partial data?